storage/s3.go

```diff
@@ -178,7 +178,13 @@ func (s *S3) Delete(ctx context.Context, path string) error {
 	input := &s3.DeleteObjectInput{Bucket: &s.cfg.Bucket, Key: &path}
 	_, err := s.s3client.DeleteObject(ctx, input)
 	if err != nil {
-		return errors.Wrapf(err, "fail to delete object %v", path)
+		var apiErr smithy.APIError
+		if stderrors.As(err, &apiErr) {
+			if apiErr.ErrorCode() == NotFoundErrCode {
+				return ObjectNotFound{}
```

Comment:
Would you mind adding the `Path` in the `ObjectNotFound`

Why not having `ObjectNotFoundError` ? :D

Reply:
> Why not having `ObjectNotFoundError` ? :D

Because it means a change in the API, so a new major version... 😓 

> Would you mind adding the `Path` in the `ObjectNotFound`

Good idea, on it

Reply:
Oooh, no problem for the first part